simulariumio/data_objects/agent_data.py

```diff
@@ -114,31 +114,33 @@ def __init__(
         self.n_subpoints = (
             n_subpoints if n_subpoints is not None else np.zeros_like(radii)
         )
-        self.subpoints = subpoints if subpoints is not None else np.zeros_like(radii)
+        self.subpoints = subpoints if subpoints is not None else np.zeros((3, 3, 0, 3))
         self.draw_fiber_points = draw_fiber_points
-        self.type_ids = type_ids
+        self.n_timesteps = n_timesteps
 
     @staticmethod
-    def _get_buffer_data_dimensions(buffer_data: Dict[str, Any]) -> Tuple[int]:
-        """ """
+    def _get_buffer_data_dimensions(buffer_data: Dict[str, Any]) -> DimensionData:
+        """
+        Get dimensions of a simularium JSON dict containing buffers
+        """
         bundle_data = buffer_data["spatialData"]["bundleData"]
         total_steps = len(bundle_data)
         max_n_agents = 0
         max_n_subpoints = 0
-        for t in range(total_steps):
-            data = bundle_data[t]["data"]
-            i = 0
+        for time_index in range(total_steps):
+            data = bundle_data[time_index]["data"]
+            buffer_index = 0
```

Comment:
just from reading this, it's hard to tell at a glance what buffer_index is indexing.  I think that `data` is just an array of numbers, that is a packed list of agents, and buffer_index keeps track of where we are in that array, right?

Reply:
yes, maybe `data_index` instead? The data is packed in it's actual buffer form at this point, I realize I'm overusing "buffer" at this point

Reply:
i'm not sure how much better the name is going to get.  I suppose it might be easier to rename `data`-->`buffer`... But maybe a small comment saying that `data` is a list of numbers being the packed agent data, might be good enough.